templates/components/engines/python/agent/tools/query_engine.py

```diff
@@ -0,0 +1,45 @@
+import os
+from typing import Any, Dict, Optional
+
+from llama_index.core.tools.query_engine import QueryEngineTool
+
+
+def create_query_engine(index, params: Optional[Dict[str, Any]] = None):
+    params = params or {}
+    top_k = int(os.getenv("TOP_K", 0))
+    if top_k != 0:
+        params["similarity_top_k"] = top_k
+    # If index is index is LlamaCloudIndex
+    # use auto_routed mode for better query results
+    if index.__class__.__name__ == "LlamaCloudIndex":
+        params["auto_routed"] = True
```

Comment:
What's actual the default?

Reply:
The default is None, which is not mentioned in the client code, but it seems to me that it'll fall back to `chunk` mode.